graphql/resolvers.go

```diff
@@ -246,7 +246,88 @@ func (r *queryResolver) Projects(ctx context.Context) (*Projects, error) {
 	return &pjs, nil
 }
 
-func (r *queryResolver) TaskTests(ctx context.Context, taskID string, sortCategory *TaskSortCategory, sortDirection *SortDirection, page *int, limit *int, testName *string, status *string) ([]*restModel.APITest, error) {
+func (r *queryResolver) PatchTasks(ctx context.Context, patchID string, sortBy *TaskSortCategory, sortDirection *SortDirection, page *int, limit *int, searchInput *string, statuses []*string) ([]*TaskResult, error) {
+	sorter := ""
+	if sortBy != nil {
+		switch *sortBy {
+		case TaskSortCategoryStatus:
+			sorter = task.StatusKey
+			break
+		case TaskSortCategoryName:
+			sorter = task.DisplayNameKey
+			break
+		case TaskSortCategoryBaseStatus:
+			// base status is not a field on the task db model; therefore sorting by base status
+			// cannot be done in the mongo query. sorting by base status is done in the resolver.
+			break
+		case TaskSortCategoryVariant:
+			sorter = task.BuildVariantKey
+			break
+		default:
+			break
+		}
+	}
+	sortDir := 1
+	if *sortDirection == SortDirectionDesc {
+		sortDir = -1
+	}
+	pageParam := 0
+	if page != nil {
+		pageParam = *page
+	}
+	limitParam := 0
+	if limit != nil {
+		limitParam = *limit
+	}
+	statusesParam := []string{}
+	if statuses != nil {
+		for _, status := range statuses {
+			statusesParam = append(statusesParam, *status)
+		}
+	}
+	taskOrVariantName := ""
+	if searchInput != nil {
+		taskOrVariantName = *searchInput
+	}
+
+	tasks, err := r.sc.FindTaskResultsByVersion(patchID, taskOrVariantName, sorter, statusesParam, sortDir, pageParam, limitParam)
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error getting patch tasks for %s: %s", patchID, err.Error()))
+	}
+	if tasks == nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("No tasks found for %s", patchID))
```

Comment:
I don't think this condition will ever be true based on current code in FindTaskResultsByVersion. If you want to report no tasks found, you will have to return nil for tasks instead of an empty task array here: https://github.com/evergreen-ci/evergreen/pull/3230/files#diff-f4f61087fe74ce0a2234b7e8f47681bfR219